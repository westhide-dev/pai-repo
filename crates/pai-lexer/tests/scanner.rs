use insta::assert_snapshot;
use pai_file::SourceFile;
use pai_lexer::Lexer;

#[test]
fn main() {
    let source_file = SourceFile::read("tests/fixtures/demo.ts").unwrap();

    let lexer = Lexer::new(source_file.source());

    let content: String = lexer.map(|unit| format!("{unit:?}\n")).collect();

    assert_snapshot!(content);
}
