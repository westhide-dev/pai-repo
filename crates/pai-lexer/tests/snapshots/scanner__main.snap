---
source: crates/pai-lexer/tests/scanner.rs
expression: content
---
Ok(Punctuator(Hash))
Ok(Punctuator(Not))
Ok(Ident(Ident { raw: "hashbang" }))
Ok(Comment(Line("... comment str")))
Ok(Comment(Block(" block comment")))
Ok(Keyword(Const))
Ok(Ident(Ident { raw: "ab" }))
Ok(Punctuator(Assign))
Ok(Punctuator(LBrace))
Ok(Punctuator(RBrace))
Ok(Punctuator(Semi))
Ok(Comment(Line(" \\u{20FFFF}")))
Ok(Keyword(Const))
Ok(Ident(Ident { raw: "c" }))
Ok(Punctuator(Assign))
Ok(Lit(Number("0o21_033553")))
Ok(Punctuator(Semi))
Ok(Keyword(Const))
Ok(Ident(Ident { raw: "d" }))
Ok(Punctuator(Assign))
Ok(Lit(Number("0.1534")))
Ok(Punctuator(Semi))
Ok(Keyword(Const))
Ok(Ident(Ident { raw: "e" }))
Ok(Punctuator(Assign))
Ok(Lit(Number("1215")))
Ok(Punctuator(Semi))
Ok(Keyword(Const))
Ok(Ident(Ident { raw: "f" }))
Ok(Punctuator(Assign))
Ok(Lit(String("hello word")))
Ok(Punctuator(Semi))

